enterprise/server/group_status/group_status_manager.go

```diff
@@ -0,0 +1,122 @@
+package group_status
```

Comment:
I am a little on the fence about adding a new package to manage a single `Group` field, since the `userdb` package is the existing place where we manage the `Group` table and it might seem somewhat arbitrary/confusing to a new reader that we have a dedicated package for one specific Group column. wdyt?

Reply:
Will address this in separate PR